### PR TITLE
 feat(evm_bridge): Add transaction pool and workers, multiple small improvements.

### PR DESCRIPTION
**Pull request by Tirka**

### CODE REVIEW

File: evm-utils/evm-bridge/Cargo.toml [hunk 1]
```diff
@@ -32,9 +32,12 @@ jsonrpc-derive = "17.1.0"
 jsonrpc-http-server = "17.1.0"
 jsonrpc-pubsub = "17.0.0"
 jsonrpc-ws-server = "17.0.0"
-num_cpus = "1.13.0"
+# num_cpus = "1.13.0"
```
Comment (vldm) [inline evm-utils/evm-bridge/Cargo.toml]: dont forget to clean commented crates.

File: evm-utils/evm-bridge/src/main.rs [hunk 2]
```diff
@@ -200,275 +203,76 @@ impl EvmBridge {
...
-                    }
-                }
-            }
-        }
-
-        let mut ix =
-            solana_evm_loader_program::send_raw_tx(self.key.pubkey(), tx, Some(self.key.pubkey()));
+    fn send_tx(&self, tx: evm::Transaction, meta_keys: HashSet<Pubkey>) -> EvmResult<Hex<H256>> {
+        let tx =
+            PooledTransaction::new(tx, meta_keys).map_err(|_e| evm_rpc::Error::Unimplemented {})?; // TODO: fix enum variant
```
Comment (vldm) [inline evm-utils/evm-bridge/src/main.rs]: Try propagate the original error.

File: evm-utils/evm-bridge/src/main.rs [hunk 3]
```diff
@@ -200,275 +203,76 @@ impl EvmBridge {
...
+    /// Returns nonce from transaction pool, on `None` if the pool doesn't contain
+    /// any transactions associated with specified sender
+    fn tx_count_in_pool(&self, sender: &Address) -> Option<U256> {
+        let pool = self.pool.lock().unwrap();
+        pool.pending_from_sender(
+            |_tx: &PooledTransaction| Readiness::Ready,
+            sender,
+            H256::zero(),
+        )
+        .map(|tx| tx.inner.nonce)
```
Comment (vldm) [inline evm-utils/evm-bridge/src/main.rs]: Looks like `max_by_key` https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key implementation

**Comment by Tirka** [comment]
Ready for review.

**Comment by Tirka** [comment]
Applied clippy suggestions related to this PR

**Comment by Tirka** [comment]
Applied clippy suggestions related to this PR

**Pull request merged**